packages/aws-cdk/lib/api/logs/register-cloudwatch-logs.ts

```diff
@@ -0,0 +1,90 @@
+import * as cxapi from '@aws-cdk/cx-api';
+import { Mode, SdkProvider, ISDK } from '../aws-auth';
+import { EvaluateCloudFormationTemplate, LazyListStackResources } from '../evaluate-cloudformation-template';
+
+// resource types that have associated CloudWatch Log Groups that should
+// _not_ be monitored
+const IGNORE_LOGS_RESOURCE_TYPES = ['AWS::EC2::FlowLog', 'AWS::CloudTrail::Trail', 'AWS::CodeBuild::Project'];
+
+// resource types that will create a cloudwatch log group
+// with a specific name if one is not provided
+const RESOURCE_TYPES_WITH_IMPLICIT_LOGS = ['AWS::Lambda::Function'];
+
+export async function registerCloudWatchLogGroups(
+  sdkProvider: SdkProvider,
+  stackArtifact: cxapi.CloudFormationStackArtifact,
+): Promise<{ env: cxapi.Environment, sdk: ISDK, logGroupNames: string[] }> {
+
+
+  const resolvedEnv = await sdkProvider.resolveEnvironment(stackArtifact.environment);
```

Comment:
Maybe let's add a `// To Do` here that this is temporary, and is waiting for https://github.com/aws/aws-cdk/pull/18277 to be merged?